### ISSUE TITLE
Assert failure(PID 396 [0x0000018c], Thread: 396 [0x018c]): Assertion failed '!"Bad type in gtNewZeroConNode"' in 'System.Runtime.Intrinsics.Vector128:AsByte(System.Runtime.Intrinsics.Vector1281[Boolean]):System.Runtime.Intrinsics.Vector1281[Byte]' during 'Morph - Global' (IL size 7)

### ISSUE THREAD

**Issue by AriNuer** [dotnet/runtime#33616]
**Job:**
[runtime-coreclr r2r-extra:20200314.1](https://dev.azure.com/dnceng/public/_build/results?buildId=559970)
**Error message:**
```
Assert failure(PID 396 [0x0000018c], Thread: 396 [0x018c]): Assertion failed '!"Bad type in gtNewZeroConNode"' in 'System.Runtime.Intrinsics.Vector128:AsByte(System.Runtime.Intrinsics.Vector1281[Boolean]):System.Runtime.Intrinsics.Vector1281[Byte]' during 'Morph - Global' (IL size 7)
File: /__w/1/s/src/coreclr/src/jit/gentree.cpp Line: 5959
Image: /root/helix/work/correlation/corerun\n\napply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f9f9f126c)
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f9f9f126c)\napply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f9f9e5ee4)
/root/helix/work/workitem/JIT/HardwareIntrinsics/General/NotSupported/NotSupported_r/NotSupported_r.sh: line 275: 396 Aborted $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code: 1
Raw output file: /root/helix/work/workitem/JIT/HardwareIntrinsics/Reports/JIT.HardwareIntrinsics/General/NotSupported/NotSupported_r/NotSupported_r.output.txt
Raw output:
BEGIN EXECUTION
in takeLock
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0\nCopyright (c) Microsoft Corporation. All rights reserved.\n\nNative image NotSupported_r.dll generated successfully.
in ReleaseLock
in takeLock
/root/helix/work/correlation/corerun NotSupported_r.dll ''
Beginning test case Vector64Zero at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro

Ending test case at 3/15/2020 4:16:53 AM
Beginning test case Vector64BooleanAsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro

Ending test case at 3/15/2020 4:16:53 AM
Beginning test case Vector64ByteAsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro
Ending test case at 3/15/2020 4:16:53 AM
Beginning test case Vector64DoubleAsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro
Ending test case at 3/15/2020 4:16:53 AM
Beginning test case Vector64Int16AsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro
Ending test case at 3/15/2020 4:16:53 AM
Beginning test case Vector64Int32AsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro
Ending test case at 3/15/2020 4:16:53 AM
Beginning test case Vector64Int64AsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro

Ending test case at 3/15/2020 4:16:53 AM
Beginning test case Vector64SByteAsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro

Ending test case at 3/15/2020 4:16:53 AM\nBeginning test case Vector64SingleAsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro
Ending test case at 3/15/2020 4:16:53 AM
Beginning test case Vector64UInt16AsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro

Ending test case at 3/15/2020 4:16:53 AM
Beginning test case Vector64UInt32AsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro

Ending test case at 3/15/2020 4:16:53 AM\nBeginning test case Vector64UInt64AsGeneric_Boolean at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro

Ending test case at 3/15/2020 4:16:53 AM
Beginning test case Vector64BooleanAsGeneric_Byte at 3/15/2020 4:16:53 AM
Random seed: 20010415; set environment variable CORECLR_SEED to this value to repro 

Ending test case at 3/15/2020 4
```
**Stack trace:**
`
at JIT_HardwareIntrinsics._General_NotSupported_NotSupported_r_NotSupported_r_._General_NotSupported_NotSupported_r_NotSupported_r_sh() in /__w/1/s/artifacts/tests/coreclr/Linux.arm64.Checked/TestWrappers/JIT.HardwareIntrinsics/JIT.HardwareIntrinsics.XUnitWrapper.cs:line 1498
`
**Details:**
https://dev.azure.com/dnceng/public/_build/results?buildId=559970&view=ms.vss-test-web.build-test-results-tab&runId=17640366&resultId=109895&paneView=debug

**Comment by sandreenko** [comment]
https://github.com/dotnet/runtime/pull/33457

**Comment by v-haren** [comment]
failed again in job: [runtime-coreclr jitstress 20200315.1](https://dev.azure.com/dnceng/public/_build/results?buildId=560455&view=ms.vss-test-web.build-test-results-tab&runId=17650012&resultId=101248&paneView=debug)

failed test:
JIT/HardwareIntrinsics/General/Vector256/Vector256_r/Vector256_r.sh
JIT/HardwareIntrinsics/General/Vector128_1/Vector128_1_r/Vector128_1_r.sh
JIT/HardwareIntrinsics/General/Vector128/Vector128_ro/Vector128_ro.sh
JIT/HardwareIntrinsics/General/Vector128/Vector128_r/Vector128_r.sh
JIT/Regression/JitBlue/GitHub_22850/GitHub_22850/GitHub_22850.sh

**Comment by BruceForstall** [comment]
@sandreenko This is a regression from https://github.com/dotnet/runtime/pull/33457? Do you have a fix in progress? Is it worth reverting https://github.com/dotnet/runtime/pull/33457 until you have a fix ready?

**Comment by sandreenko** [comment]
@sandreenko This is a regression from #33457? Do you have a fix in progress? Is it worth reverting #33457 until you have a fix ready?


Yes, I am going to publish it today.
